### PR TITLE
[#158389284] update date format with locale on message preview

### PR DESCRIPTION
**Pull request by valentinacala**
The pr introduces the locale check for setting the date format on the message preview:
* MM/DD for english
* DD/MM for italian

### CODE REVIEW

File: ts/utils/convertDateToWordDistance.ts [hunk 1]
```diff
@@ -20,7 +22,12 @@ export function convertDateToWordDistance(
     return lastDayLabel;
   } // 1 day < distance < 365 days, current year
   else if (distance > 1 && distance < 365) {
-    return format(date, "DD/MM");
+    const locale = I18n.currentLocale();
```
Comment (fpersico) [inline ts/utils/convertDateToWordDistance.ts]: The problem with this solution is that we have to change the code each time we add a new language. I think you can use the utility function exported here https://github.com/teamdigitale/io-app/blob/master/ts/utils/dates.ts#L26.

**Comment by cloudify** [comment]
Please ignore Hound errors, I've disabled it

File: ts/utils/convertDateToWordDistance.ts (hunk 1 above)
Comment (fpersico) [reply ts/utils/convertDateToWordDistance.ts]: Please check the comment above.

Comment (valentinacala) [reply ts/utils/convertDateToWordDistance.ts]: Thank you for feedback. I'm looking for it but it seems date-fns doesn't provide a function which makes the date format depends on the passed locale. I could implement the function in date.ts and pass it to convertDateToWordDistance.ts . What do you think?

Comment (fpersico) [reply ts/utils/convertDateToWordDistance.ts]: Another possible solution https://github.com/fnando/i18n-js#date-formatting

File: ts/@types/native-base-types.d.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ declare module "native-base-shoutem-theme" {
   /**
    * The connectStyle function does not use forwardRef.
    * To get a ref to the wrapped component you have to use _root.
-   * 
+   *
```
Comment (cloudify) [inline ts/@types/native-base-types.d.ts]: what happened here?

File: ts/@types/react-native-touch-id.d.ts [hunk 3]
```diff
@@ -1,112 +1,111 @@
-declare module 'react-native-touch-id' {
+declare module "react-native-touch-id" {
```
Comment (cloudify) [inline ts/@types/react-native-touch-id.d.ts]: what happened here?

File: ts/utils/dates.ts [hunk 4]
```diff
@@ -23,12 +23,26 @@ export function formatDateAsReminder(
   return dateFnsFormat(date, "YYYY-MM-DDTHH:mm:ss.SSS[Z]");
 }
 
+export function formatDateAsLocal(
```
Comment (cloudify) [inline ts/utils/dates.ts]: Please add a comment that explain the purpose of this

File: ts/utils/dates.ts [hunk 5]
```diff
@@ -23,12 +23,26 @@ export function formatDateAsReminder(
...
+  date: Date,
+  includeYear: boolean = false
+): ReturnType<typeof dateFnsFormat> {
+  enum LocaleFormat {
+    "" = "MM/DD",
+    "en" = "MM/DD",
```
Comment (cloudify) [inline ts/utils/dates.ts]: these formats should go [here](https://github.com/teamdigitale/io-app/blob/master/locales/it/index.yml#L5)

File: ts/@types/react-native-touch-id.d.ts (hunk 3 above)
Comment (valentinacala) [reply ts/@types/react-native-touch-id.d.ts]: same as previous comment

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#158389284](https://www.pivotaltracker.com/story/show/158389284): Come CIT quando visualizzo i messaggi (lista e dattaglio) voglio che il campo "data" sia correttamente visualizzato secondo la Timezone del mio dispositivo. 

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/@types/native-base-types.d.ts (hunk 2 above)
Comment (cloudify) [reply ts/@types/native-base-types.d.ts]: revert this change please

File: ts/@types/react-native-touch-id.d.ts (hunk 3 above)
Comment (cloudify) [reply ts/@types/react-native-touch-id.d.ts]: revert this change please

File: ts/utils/convertDateToWordDistance.ts [hunk 6]
```diff
@@ -1,8 +1,9 @@
 import { differenceInCalendarDays, format } from "date-fns";
+import { formatDateAsLocal } from "./dates";
+
 /**
- * This function converts the distance from now to date in : H.mm, yesterday, D/MM/YY and DD/MM
- *  TODO: Make date consistent with the timezone
- *  https://www.pivotaltracker.com/story/show/158389284
+ * This function converts the distance from now to date in :
+ * H.mm, yesterday, MM/DD (or DD/MM) and MM/DD/YY (or DD/MM/YY) depending on user geography
```
Comment (cloudify) [inline ts/utils/convertDateToWordDistance.ts]:
```suggestion
 * H.mm, yesterday, MM/DD (or DD/MM) and MM/DD/YY (or DD/MM/YY) depending on the system locale
```

File: ts/utils/dates.ts [hunk 7]
```diff
@@ -23,12 +23,28 @@ export function formatDateAsReminder(
   return dateFnsFormat(date, "YYYY-MM-DDTHH:mm:ss.SSS[Z]");
 }
 
+/**
+ *
+ * It provides the format of the date depending on the geography (DD/MM or MM/DD as default)
```
Comment (cloudify) [inline ts/utils/dates.ts]:
```suggestion
 * It provides the format of the date depending on the system locale (DD/MM or MM/DD as default)
```

**Pull request merged**